ls1intum/Artemis
# `Discussion`: Add button to show / hide post replies

### Checklist
#### General


#### Client

### Motivation and Context


User feedback lead to the change request that the previous UI for toggling the answers of a post is not intuitive / learnable.
With this new UI we strive for better understanding and consistency between posts without answers (show `Start Discussion` button) and with answers (did not show any button).

### Description

There will always be a button below the post content. 
**We distinguish following use cases:**
1. There is no answer yet: `Start Discussion` Button visible in post footer, clicking it will open the modal
2. There are answers, but not expanded yet: `Show X Reply` Button visible with answer count, clicking it will expand the answers (turning to state described in 3.)
3. There are answers, they are expanded: `Reply Now` Button visible, clicking it will open the modal, _AND_ `Hide Answers` Button visible, clicking it will hide the answers (turning back to state described in 2.)

### Steps for Testing


1. Log in to Artemis
2. Navigate course discussion overview / lecture page or exercise page
3. Find a post to reenact the use cases described in the **Description** section
4. Write a new post and check if in the similar posts shown when providing a post title, there are no buttons.

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage






### Screenshots


Old UI for posts without, with 1, with 2 answers:


https://user-images.githubusercontent.com/41366522/139492154-669ad2bb-bafb-4dd9-a5db-ee42ba5ffd86.mov



New UI for posts without, with 1, with 2 answers:

https://user-images.githubusercontent.com/41366522/139491398-e406163a-836c-472e-b635-9606d5842e8a.mov
